src/tools/miri/tests/pass/async-drop.rs

```diff
@@ -11,9 +11,21 @@ use core::mem::{self, ManuallyDrop};
 use core::pin::{pin, Pin};
 use core::task::{Context, Poll, Waker};
 
-async fn test_async_drop<T>(x: T) {
+async fn test_async_drop<T>(x: T, _size: usize) {
     let mut x = mem::MaybeUninit::new(x);
     let dtor = pin!(unsafe { async_drop_in_place(x.as_mut_ptr()) });
+
+    // FIXME(zetanumbers): This check fully depends on the layout of
+    // the coroutine state, since async destructor combinators are just
+    // async functions.
+    #[cfg(target_pointer_width = "64")]
+    assert_eq!(
+        mem::size_of_val(&*dtor),
+        _size,
+        "sizes did not match for async destructor of type {}",
+        core::any::type_name::<T>(),
+    );
```

Comment:
Initially I've put it there to test it out on a 32bit target without needing to cross-compile. It worked, but then I thought it should be easier to maintain this test if those sizes are represented with a number of bytes instead of a number of "words" or pointers and restricted it to 64bit targets. 

Also two async drop tests, one from tests/ui and other from miri are mostly each other's copy.

Sorry for the confusion. Should I disable this check with `cfg(miri)` or just remove it from the code in miri test?

Reply:
This is testing a compiler property, nothing about Miri or the interpreter, so having the test in Miri is a bit odd. Since a similar size test already exists in rustc, having to maintain a copy of that in Miri does not seem great.

So ideally the Miri test would just check that Miri can successfully execute async drop, but would not do size checks.

> Should I disable this check with cfg(miri)

This makes little sense as this test is only executed with Miri.

Reply:
> This makes little sense as this test is only executed with Miri.

But then I wouldn't be able to just copy one test into another without modifications. Well ok. I should also add notes to them about the other test.

Reply:
In the future if the sizes change, should the test then be re-copied every time? That also doesn't seem great.

Reply:
Changed in 98c0d7b7379b427dc8c18b85b34ff5a89cf06118